### PR TITLE
Slightly re-structuring BreweryDB client, adding some tests

### PR DESCRIPTION
**Pull request by hobbeswalsh**
@michaelporter 

Hey there, thanks for getting this project off the ground. I've switched around how the client works a little bit, added a few structs for JSON decoding, and added some unit tests. Let me know if you think the changes look good.

Thanks!

### CODE REVIEW

**Comment by hobbeswalsh** [comment]
Oh yeah, and I've added a `.travis.yml` file as well, so we can have continuous testing on Travis-CI.

**Comment by michaelporter** [comment]
This looks awesome - I'll take a closer look at it this evening!

**Comment by hobbeswalsh** [comment]
@michaelporter Thanks. My apologies if I stomped on your original intentions for the library's public API; I figured `NewClient()` was pretty idiomatic, and then calling methods on that client that correspond to the REST endpoints makes sense, but let me know if you want to structure it differently!

Thanks!

File: README.md [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
+    package main
+
     import (
     	"github.com/michaelporter/brewerydb"
     )
+    
+    c := brewerydb.NewClient("myApiKey")
+
+    // I want a random beer.
+    beer, err := c.RandomBeer()
```
Comment (michaelporter) [inline README.md]: Looks like RandomBeer() requires a BeerQuery param; can you update the README to illustrate that?

**Comment by michaelporter** [comment]
No worries about stomping - this was my first project in Go, so the design was a second-class citizen (at best).  I like the direction you've taken it in

File: README.md [hunk 2]
```diff
@@ -1,22 +1,29 @@
...
-    func getBeers() {
-      brewerydb.SetApiKey("your-api-key")
-      
-      queryParams := map[string]string{
-      	'abv': '6',
-      }
-      res, err := brewerydb.Get("/beers", queryParams)
-    }
+    // I want a list of breweries near some zip code
+    lq = brewerydb.LocationQuery{}
```
Comment (michaelporter) [inline README.md]: this example doesn't run, given that lq needs to be instantiated `lq := brewerydb.LocationQuery{}`

Comment (hobbeswalsh) [inline README.md]: oops! moving from other languages to Golang kinda sucks for variable assignment :)

File: README.md (hunk 1 above)
Comment (hobbeswalsh) [inline README.md]: thanks, fixed.

**Pull request merged**